packages/features/voting-context.tsx

```diff
@@ -188,10 +194,75 @@ function VotingCtxProvider({ children }: ChildrenProps) {
     }, 1000);
   }
 
-  function handleVotingEnd() {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  async function handleVotingEnd() {
     // TODO save result to db
     setEndMessage('Eleição Encerrada');
     setIsVoting(false);
+    await postResult();
+  }
+
+  async function postResult() {
+    const votes = await getVotes(currentElectionId);
+    const configDatabase = await getConfiguration();
+
+    const votedSet = new Set();
+    const voteCounts: any = {};
+
+    const election: AvailableElections | undefined =
+      availableElections.results?.find((election) => {
+        return election.electionId === currentElectionId;
+      });
+
+    if (!election) {
+      return;
+    }
+
+    votes.forEach((vote) => {
+      const code = vote.code;
+      if (!isNaN(parseFloat(code))) {
+        if (votedSet.has(code)) {
+          voteCounts[code] += 1;
+        } else {
+          votedSet.add(code);
+          voteCounts[code] = 1;
+        }
+      }
+    });
+
+    const [[winnerCode, winnerVotes], [loserCode, loserVotes]]: any =
+      Object.entries(voteCounts).sort((a: any, b: any) => b[1] - a[1]);
+
+    const [{ members: winnerMembers, name: winnerName }] = partyList.filter(
+      (party) => party.code === winnerCode,
+    );
+    const [{ members: loserMembers, name: loserName }] = partyList.filter(
+      (party) => party.code === loserCode,
+    );
```

Comment:
Maybe we can just turn this part into a function for countingResult

Reply:
Fiz um refactor da função de contagem de votos e isolei o objeto de resultado final para o POST no notion.